pkg/skaffold/initializer/init_test.go

```diff
@@ -59,3 +63,146 @@ func TestPrintAnalyzeJSON(t *testing.T) {
 		})
 	}
 }
+
+func TestWalk(t *testing.T) {
+	emptyFile := []byte("")
+	tests := []struct {
+		name                string
+		filesWithContents   map[string][]byte
+		expectedConfigs     []string
+		expectedDockerfiles []string
+		force               bool
+		err                 bool
+	}{
+		{
+			name: "should return correct k8 configs and dockerfiles",
+			filesWithContents: map[string][]byte{
+				"config/test.yaml":  emptyFile,
+				"k8pod.yml":         emptyFile,
+				"README":            emptyFile,
+				"deploy/Dockerfile": emptyFile,
+				"Dockerfile":        emptyFile,
+			},
+			force: false,
+			expectedConfigs: []string{
+				"config/test.yaml",
+				"k8pod.yml",
+			},
+			expectedDockerfiles: []string{
+				"Dockerfile",
+				"deploy/Dockerfile",
+			},
+			err: false,
+		},
+		{
+			name: "should skip hidden dir",
+			filesWithContents: map[string][]byte{
+				".hidden/test.yaml":  emptyFile,
+				"k8pod.yml":          emptyFile,
+				"README":             emptyFile,
+				".hidden/Dockerfile": emptyFile,
+				"Dockerfile":         emptyFile,
+			},
+			force: false,
+			expectedConfigs: []string{
+				"k8pod.yml",
+			},
+			expectedDockerfiles: []string{
+				"Dockerfile",
+			},
+			err: false,
+		},
+		{
+			name: "should not error when skaffold.config present and force = true",
+			filesWithContents: map[string][]byte{
+				"skaffold.yaml": []byte(`apiVersion: skaffold/v1beta6
+kind: Config
+deploy:
+  kustomize: {}`),
+				"config/test.yaml":  emptyFile,
+				"k8pod.yml":         emptyFile,
+				"README":            emptyFile,
+				"deploy/Dockerfile": emptyFile,
+				"Dockerfile":        emptyFile,
+			},
+			force: true,
+			expectedConfigs: []string{
+				"config/test.yaml",
+				"k8pod.yml",
+			},
+			expectedDockerfiles: []string{
+				"Dockerfile",
+				"deploy/Dockerfile",
+			},
+			err: false,
+		},
+		{
+			name: "should  error when skaffold.config present and force = false",
+			filesWithContents: map[string][]byte{
+				"config/test.yaml":  emptyFile,
+				"k8pod.yml":         emptyFile,
+				"README":            emptyFile,
+				"deploy/Dockerfile": emptyFile,
+				"Dockerfile":        emptyFile,
+				"skaffold.yaml": []byte(`apiVersion: skaffold/v1beta6
+kind: Config
+deploy:
+  kustomize: {}`),
+			},
+			force:               false,
+			expectedConfigs:     nil,
+			expectedDockerfiles: nil,
+			err:                 true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			rootDir, err := ioutil.TempDir("", "test")
```

Comment:
There's an util to do that: `testutil.NewTempDir(t)`. It should make `createDirStructure` easier to write.